docs/migration-guide/w3c-vc-v2.md

```diff
@@ -0,0 +1,565 @@
+  mandatoryPointers: [
+    '/credentialSubject/id',
+    '/credentialSubject/name',
+    '/credentialSubject/degree'
+  ]
+});
+```
+
+##### Step 2: Derive Selective Disclosure Credential
+
+```typescript
+import { deriveW3C } from '@trustvc/trustvc';
+
+// Create job application VC (hide sensitive info)
+const jobApplicationVC = await deriveW3C(fullCredential, [
+  '/credentialSubject/graduationYear'
+  // Reveals: id, name, degree (mandatory) + graduationYear
+  // Hides: studentId, gpa, dateOfBirth
+]);
+
+// Create scholarship application VC (show academic performance)
+const scholarshipVC = await deriveW3C(fullCredential, [
+  '/credentialSubject/gpa',
+  '/credentialSubject/graduationYear'
+  // Reveals: id, name, degree (mandatory) + gpa + graduationYear
+  // Hides: studentId, dateOfBirth
+]);
+```
+
+##### Step 3: Verify the Derived Credential
+
+```typescript
+import { verifyW3CSignature } from '@trustvc/trustvc';
+
+// ❌ WRONG: Cannot verify original credential with ECDSA-SD-2023
+// const result = await verifyW3CSignature(fullCredential); // This will fail
+
+// ✅ CORRECT: Verify the derived credential
+const jobVerification = await verifyW3CSignature(jobApplicationVC);
+const scholarshipVerification = await verifyW3CSignature(scholarshipVC);
+
+console.log('Job Application Valid:', jobVerification.verified);
+console.log('Scholarship Application Valid:', scholarshipVerification.verified);
+```
+
+> **⚠️ Important**: Unlike BBS+ credentials, ECDSA-SD-2023 credentials **must be derived before verification**. You cannot verify the original full credential directly if you want selective disclosure.
+
+#### Best Practices for Selective Disclosure
+
+1. **Plan Mandatory Pointers Carefully**: Include fields that should always be visible (e.g., subject ID, credential type)
+2. **Use Meaningful Field Selection**: Only reveal fields necessary for the specific use case
+3. **Always Derive Before Verification**: Remember that ECDSA-SD-2023 requires derivation before verification
+4. **Test Different Disclosure Scenarios**: Ensure your selective disclosure works for various use cases
+5. **Document Your Disclosure Policies**: Clearly communicate what fields are mandatory vs. selectable
+
+### 5. Verification Migration
+
+The TrustVC library provides seamless verification through the `verifyDocument` function, which offers **backward compatibility** and works with both v1.1 and v2.0 credentials automatically.
+
+#### Unified Verification with verifyDocument
+
+```typescript
+import { verifyDocument } from '@trustvc/trustvc';
+
+// ✅ Works with v1.1 BBS+ credentials
+const v1Result = await verifyDocument(signedV1BBSCredential);
+
+// ✅ Works with derived v2.0 ECDSA-SD-2023 credentials
+const v2Result = await verifyDocument(derivedV2ECDSACredential);
+
+console.log('v1.1 Verification:', v1Result.verified);
+console.log('v2.0 Verification:', v2Result.verified);
+```
```

Comment:
How about original document that havent been derived yet. I tot our verifier function can handle that after the discussion we had?